polygeist/tools/cgeist/driver.cc

```diff
@@ -606,16 +621,17 @@ static int finalize(mlir::MLIRContext &context,
       module->dump();
     });
 
-    mlir::PassManager pm2(&context);
-    if (SCFOpenMP) {
+    mlir::PassManager pm2(&context, ModuleTy::OperationT::getOperationName());
+    if (SCFOpenMP &&
+        std::is_same<mlir::ModuleOp, typename ModuleTy::OperationT>::value) {
```

Comment:
At some point we will want to support OpenMP offloading (OpenMP device function & target regions) in addition to SYCL. For now I am OK with this change which causes the optimization to be run only for the CPU ModuleOp and not in the GPUModule but something to keep in mind.

Reply:
This pass does not work with `GPUModuleOp`. If we port it, there'll be no difference between both modules here.

Reply:
ok